server/src/main/java/org/eclipse/openvsx/security/AuthUserFactory.java

```diff
@@ -0,0 +1,20 @@
+package org.eclipse.openvsx.security;
+
+import org.springframework.security.oauth2.core.user.OAuth2User;
+import org.springframework.stereotype.Service;
+
+@Service
+public class AuthUserFactory {
+
+    public AuthUser createAuthUser(String providerId, OAuth2User oauth2User) {
+        return new DefaultAuthUser(
+            oauth2User.getName(),
+            oauth2User.getAttribute("avatar_url"),
+            oauth2User.getAttribute("email"),
+            oauth2User.getAttribute("name"),
+            oauth2User.getAttribute("login"),
+            providerId,
+            oauth2User.getAttribute("html_url")
+        );
+    }
+}
```

Comment:
@HeikoBoettger this class should make it easier to customize `AuthUser` instances. Right now it is hardcoded but you can later make this factory depend on some Spring configuration to customize attribute access.

Reply:
@paul-marechal I am not so familiar with spring, do you know how to map the "login" in the spring configuration? I only know about the "spring.security.oauth2.client.provider.my-oauth-provider.user-name-attribute=name". May be short hint somewhere in the openvsx documentation on how to customize the attribute access would be good.

Great work by the way.